Userland/Applications/PixelPaint/Tools/PolygonalSelectTool.cpp

```diff
@@ -142,12 +142,14 @@ void PolygonalSelectTool::on_second_paint(Layer const* layer, GUI::PaintEvent& e
     for (size_t i = 0; i < m_polygon_points.size() - 1; i++) {
         auto preview_start = editor_stroke_position(m_polygon_points.at(i), 1);
         auto preview_end = editor_stroke_position(m_polygon_points.at(i + 1), 1);
-        painter.draw_line(preview_start, preview_end, Color::Black, 1);
+        painter.draw_line(preview_start, preview_end, Color::Black, 3);
+        painter.draw_line(preview_start, preview_end, Color::White, 1);
     }
 
     auto last_line_start = editor_stroke_position(m_polygon_points.at(m_polygon_points.size() - 1), 1);
     auto last_line_stop = editor_stroke_position(m_last_selecting_cursor_position, 1);
-    painter.draw_line(last_line_start, last_line_stop, Color::Black, 1);
+    painter.draw_line(last_line_start, last_line_stop, Color::Black, 3);
+    painter.draw_line(last_line_start, last_line_stop, Color::White, 1);
 }
```

Comment:
Kinda a nitpick: You'll get the same look as the lasso tool if you move all of this line drawing code into a lambda/function, then can be used to first draw a black outline, then a white outline (right now does not quite work at the endpoints).

So you'd end up with two calls:
```
draw_preview_lines(Color::Black, 3);
draw_preview_lines(Color::White, 1);
```

Reply:
Refactored into a lambda. The endpoints now appear as expected.